src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -29,66 +30,113 @@ internal abstract partial class AbstractAddParameterCheckCodeRefactoringProvider
                 {
-                    return ImmutableArray<CodeAction>.Empty;
+                    listOfParametersOrdinals.Add(parameter.Ordinal);
                 }
+            }
 
-                foreach (var statement in blockStatementOpt.Operations)
+            if (!listOfParametersOrdinals.Any())
+            {
+                return ImmutableArray<CodeAction>.Empty;
+            }
+
+            // Great.  The list has parameters that need null checks. Offer to add null checks for all.
+            return ImmutableArray.Create<CodeAction>(new MyCodeAction(
+                 FeaturesResources.Add_null_checks_for_all_parameters,
+                     c => DocumentRetriever(document, functionDeclaration, blockStatementOpt, listOfParametersOrdinals, c)));
+
+            // DocumentRetriever updates all necessary variables per each parameter and calls for AddNullCheckAsync
+            async Task<Document> DocumentRetriever(
+                Document document, SyntaxNode functionDeclaration, IBlockOperation blockStatementOpt, List<int> listOfParametersOrdinals, CancellationToken cancellationToken)
+            {
+                foreach (var index in listOfParametersOrdinals)
                 {
-                    if (IsIfNullCheck(statement, parameter))
+                    // Updates functionDeclaration and uses it to get the first valid ParameterNode using the ordinals (index).
+                    var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                    var token = root.FindToken(position);
+                    var firstparameterNode = GetParameterNode(token, position);
+                    functionDeclaration = firstparameterNode.FirstAncestorOrSelf<SyntaxNode>(IsFunctionDeclaration);
+
+                    var currentNode = GetParameterNodeAtIndex(index, functionDeclaration);
+                    if (currentNode == null)
                     {
-                        return ImmutableArray<CodeAction>.Empty;
+                        continue;
                     }
 
-                    if (ContainsNullCoalesceCheck(
-                            syntaxFacts, semanticModel, statement,
-                            parameter, cancellationToken))
+                    var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                    var parameter = (IParameterSymbol)semanticModel.GetDeclaredSymbol(currentNode, cancellationToken);
+                    var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+
+                    if (!CanOfferRefactoring(functionDeclaration, semanticModel, syntaxFacts, cancellationToken, out blockStatementOpt))
+                    {
+                        continue;
+                    }
+
+                    // If parameter is a string, default check would be IsNullOrEmpty - updates document
```

Comment:
why?

Reply:
We had UX checks and based on telemetry we found that this is the most common check for strings. What do you think about it?

Reply:
That's totally reasonable.  But the comment should explain as much.  It would not be something someone could tell by looking at the code.  Thanks! :)